### PR TITLE
running no longer takes stamina,humans have less buffer, weeds slow you down minorly

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Running no longer takes stam, but marines still stop running if they run out of stamina buffer and start taking stamina loss.
fixes <https://github.com/tgstation/TerraGov-Marine-Corps/issues/6946> by removing the check that forces you to walk when buckling, not really needed anymore
Humans have 20 less buffer, moved to overall stamina pool instead
Weeds slow humans down minorly.

## Why It's Good For The Game
Running stamina is a terrible gameplay mechanic that barely helps the game and makes marines massively defensive as they're slow as a rock in usual play, this makes for boring, uninteresting gameplay
Marines being faster opens up a better game for everyone as the game is able to be more aggressive

## Changelog
:cl:
balance:  Running no longer takes stamina, humans have less buffer before forced to walk.
balance: As running no longer takes stamina, marines are now slown down by weeds minorly again, not as much as it used to be as xenomorphs are sped up on weeds.
/:cl:

### CODE REVIEW

**Comment by usnpeepoo** [comment]
make weeds slowdown marines too

**Comment by Walarks** [comment]
For how long this has been discussed, I think it's easier to test this PR for a few rounds and be done with this question for the near future.

**Comment by David-Stormwell** [comment]
I remember why stamina was first made.
Because you had me running down sentinels with light armor pre-bump slash with little effort.
Because you had sentinels and queens with their hallos attacks, causing you the player to throw up.

The game has changed significantly since then, we now have acid turrets and acid wells to stop those lone marines from making a daring solo flank and get away with it, runners which can slowly wear you down with slashes and evasion, etc.

Though it will cause problems to arise once again in lowpop, as that what the mechanic was designed for, and the fact because of it certain weapons will be quite stronger once again.
I'd like people to consider it, perhaps some other things will have to be slightly adjusted to compasate, but overall it may improve life slightly.

**Comment by Losenis** [comment]
I can predict boiler will be pure pain to play as given they're S O F U C K I N G S L O W

**Comment by BraveMole** [comment]
> I can predict boiler will be pure pain to play as given they're S O F U C K I N G S L O W

Well boiler is easy as fuck to play, maybe it will make it more interesting?

**Comment by Wayland-Smithy** [comment]
Not a fan if this makes stamina damage null for punishing over-extended sprinting marines, and you failed to make up for that. Would not having a full stamina tank all the time negate almost all stam damage?

Queen screech and Crusher toss for starters: 
https://github.com/tgstation/TerraGov-Marine-Corps/blob/7fe136f2ff04b7a36bbbd348fb37d64ac901af2a/code/modules/mob/living/carbon/human/human_defense.dm#L423
https://github.com/tgstation/TerraGov-Marine-Corps/blob/7fe136f2ff04b7a36bbbd348fb37d64ac901af2a/code/modules/mob/living/carbon/xenomorph/castes/crusher/abilities_crusher.dm#L149

**Comment by David-Stormwell** [comment]
> 
> 
> Not a fan if this makes stamina damage null for punishing over-extended sprinting marines, and you failed to make up for that. Would not having a full stamina tank all the time negate almost all stam damage? Even if you manage to 0 it out they'd still be able to run too lmao.
> 
> Queen screech and Crusher toss for starters:
> 
> https://github.com/tgstation/TerraGov-Marine-Corps/blob/7fe136f2ff04b7a36bbbd348fb37d64ac901af2a/code/modules/mob/living/carbon/human/human_defense.dm#L423
> 
> https://github.com/tgstation/TerraGov-Marine-Corps/blob/7fe136f2ff04b7a36bbbd348fb37d64ac901af2a/code/modules/mob/living/carbon/xenomorph/castes/crusher/abilities_crusher.dm#L149

This is why I said this "I'd like people to consider it, perhaps some other things will have to be slightly adjusted to compensate, but overall it may improve life slightly." because that's what this would mean. A readjustment on Xeno abilities and perhaps even guns to compasate.

**Comment by Wayland-Smithy** [comment]
So in the meantime just let it be cbt for xenos like the OP lasgun merge + no FF TM?

**Comment by Al-1ce** [comment]
> Running stamina is a terrible gameplay mechanic that barely helps the game and makes marines massively defensive as they're slow as a rock in usual play, this makes for boring, uninteresting gameplay
> Marines being faster opens up a better game for everyone as the game is able to be more aggressive

It doesn't seem you've taken into account how _unrestricted_ speed changes the balance of the game. I don't really any reasoning for this outwards of just wanting to make the game easier. This was already experienced after the jetpack PR.

I can't seriously imagine balance in a situation where the vast majoriety of marines will outspeed most xenos, minding that some already do. Marines are intended to be aggressive and with teamwork can shatter the defences of xenos, having what is essentially **another** crutch to make that easier is just poor play on all sides of the matter.

**Comment by David-Stormwell** [comment]
> So in the meantime just let it be cbt for xenos like the OP lasgun merge + no FF TM?

Comment on the fucking Lasgun PR if you think it's OP, I don't have time to waste anymore to observe every round.
And since we are on that topic, if we are TMing game changing things like no FF we might as well also test out other theories.

**Comment by Wayland-Smithy** [comment]
I'm just pointing out how one side seems to be benefitting a lot more often than the other when it comes to these feature additions and "theories" lately.

How about backing a PR to remove all stam damage from xenos and give them something better first then make it not matter for marines?

**Comment by vilereaver41** [comment]
Did you compensate for those xenos that cant outrun marines? A boiler or spitter will be doomed and dont have eyes in the back of their head from solo marinea sneaking up on them

**Comment by Knuff-les** [comment]
> Did you compensate for those xenos that cant outrun marines? A boiler or spitter will be doomed and dont have eyes in the back of their head from solo marinea sneaking up on them. This is is to big of a change not to mention one that makes no sense. A human doesent have infinite stamina there is a reason it was put in to begin with.
> 
> Marines having a change like this to help their aggressive pushes and not have time to build up a position is stupid. I havent played in 4 months and come to see wierd prs like this lol

marines having more tools to help their aggressive pushes is bad how again? xenos get atleast 30 minutes to prep defenses, and currently, marines are stuck inside FOB for thirty minutes till they EVAC, and what does human not having unlimited stamina irl have to do here? this is a game.

**Comment by David-Stormwell** [comment]
> 
> 
> I'm just pointing out how one side seems to be benefitting a lot more often than the other when it comes to these feature additions and "theories" lately.
> 
> How about backing a PR to remove all stam damage from xenos and give them something better first then make it not matter for marines?

Sure, do so.
I'd be glad for a better change! As long as it is _better_.

**Comment by Tupinambis** [comment]
This change is pretty extreme in isolation and entirely destroys xeno movespeed balance. It also makes walkspeed kind of pointless.

Perhaps this would be a good idea with many other changes but on its own it's a massive buff to marines and is going to make it much harder to slow pushes or chase ungas down or actually hit ungas to begin with. It also lets marines chase down and kill xenos much easier.

File: code/datums/components/stamina_behavior.dm [hunk 1]
```diff
@@ -25,7 +25,6 @@
 	if(stamina_state == STAMINA_STATE_ACTIVE)
 		return
 	stamina_state = STAMINA_STATE_ACTIVE
-	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/on_move_run)
```
Comment (SplinterGP) [inline code/datums/components/stamina_behavior.dm:28]: add this back

File: code/datums/components/stamina_behavior.dm [hunk 2]
```diff
@@ -36,18 +35,6 @@
...
-/datum/component/stamina_behavior/proc/on_move_run(datum/source, atom/oldloc, direction, Forced)
-	SIGNAL_HANDLER
-	if(Forced)
-		return
-	var/mob/living/stamina_holder = parent
-	if(oldloc == stamina_holder.loc)
-		return
-	stamina_holder.adjustStaminaLoss(1)
-	if(stamina_holder.staminaloss >= 0)
-		stamina_holder.toggle_move_intent(MOVE_INTENT_WALK)
```
Comment (SplinterGP) [inline code/datums/components/stamina_behavior.dm:48]: add the proc back with this bit here

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Comment by SplinterGP** [comment]
you should turn to walk incase you have no stamina since a xeno made it run out of it.

**Comment by EgorDinamit** [comment]
Marines are now literally unkillable machines. Being unable to run 24/7 was literally the only thing stopping them from winning the game in 5 minutes.

**Comment by JackTheJackhammer** [comment]
This is nice, having to click alt to move quickly is kinda annoying but really, we should look into the balance and the time it takes to go cross map and to strategic points with infinite run. May kinda be broken.

**Comment by SplinterGP** [comment]
> 
> 
> This is nice, having to click alt to move quickly is kinda annoying but really, we should look into the balance and the time it takes to go cross map and to strategic points with infinite run. May kinda be broken.

you know there is a button that toggles it right ? default is 5

**Review by SplinterGP — APPROVED** [review]
code good

**Comment by vilereaver41** [comment]
> > Did you compensate for those xenos that cant outrun marines? A boiler or spitter will be doomed and dont have eyes in the back of their head from solo marinea sneaking up on them. This is is to big of a change not to mention one that makes no sense. A human doesent have infinite stamina there is a reason it was put in to begin with.
> > Marines having a change like this to help their aggressive pushes and not have time to build up a position is stupid. I havent played in 4 months and come to see wierd prs like this lol
> 
> marines having more tools to help their aggressive pushes is bad how again? xenos get atleast 30 minutes to prep defenses, and currently, marines are stuck inside FOB for thirty minutes till they EVAC, and what does human not having unlimited stamina irl have to do here? this is a game.

oh so because some matches marines arent able to push out of FOB and get outplayed this is a reason to make them super human with infinite stamina? lol your logic is flawed, you guys have mortars, sadars, nades, airstrikes.

**Comment by Knuff-les** [comment]
> > > Did you compensate for those xenos that cant outrun marines? A boiler or spitter will be doomed and dont have eyes in the back of their head from solo marinea sneaking up on them. This is is to big of a change not to mention one that makes no sense. A human doesent have infinite stamina there is a reason it was put in to begin with.
> > > Marines having a change like this to help their aggressive pushes and not have time to build up a position is stupid. I havent played in 4 months and come to see wierd prs like this lol
> > 
> > 
> > marines having more tools to help their aggressive pushes is bad how again? xenos get atleast 30 minutes to prep defenses, and currently, marines are stuck inside FOB for thirty minutes till they EVAC, and what does human not having unlimited stamina irl have to do here? this is a game.
> 
> oh so because some matches marines arent able to push out of FOB and get outplayed

not “some”, most. I play the game, I’m not a neverplayer looking at winrates, lol. Most matches the shutters drop, we fight over the same strip of land, get pushed back into FOB and back up into space.

>this is a reason to make them super human with infinite stamina? 

Yes. You’re acting like this is something that absolutely cannot be balanced, which is wrong. You could increase slowdown on weeds, increase armor slowdown, increase stun times etc etc, the point is, it’s not something that absolutely couldn’t work within the design of the game.

>lol your logic is flawed, you guys have 

>mortars
Not an aggressive tool
>sadars
Not available on a consistent basis
> nades
Hinders as much as they help
>airstrikes
see two above

It’s crazy how marines have all these bullshit tools and they still lose more often, lol. It’s not that there is something wrong with balance, right? It has to all be a skill issue, right?

**Comment by Losenis** [comment]
>mortars, sadars, nades, airstrikes

You do realize these things can kill way more marines than xenos

**Comment by JackTheJackhammer** [comment]
Any form of explosive/AOE support is more harmful to marines than anything really

**Comment by Losenis** [comment]
Also all those tools require at least one player that is competent, willing, and able to use them properly, as well as everyone playing in a manner that supports them. Caves invalidates two of them already, and for every instance of CAS rocketing marines and cades there's another instance of marines going out of their way to run towards the green dot just for having seen a beno close.

**Comment by Surrealaser** [comment]
This is a joke PR, and has no business being TMed when I can't even get a TM to fix things that are obviously shit like the Carrier caste and Transvitox

**Comment by Losenis** [comment]
If one PR not getting TMd was justification for another PR not getting TMd, there'd be no TMs period.

**Comment by Wayland-Smithy** [comment]
Nice meme. 

Seems to make snowballs even worse somehow with light/med armor humans out running most xeno castes now, so if they don't run down the entire hive at shutters open all the overconfident robustos die thinking they can avoid death with their shotgun PAS and the fob falls shortly after with ~20 marines running back to ship hoping ERT spam works.

**Comment by Knuff-les** [comment]
> Nice meme.
> 
> Seems to make snowballs even worse somehow with light/med armor humans out running most xeno castes now, so if they don't run down the entire hive at shutters open all the overconfident robustos die thinking they can avoid death with their shotgun PAS and the fob falls shortly after with ~20 marines running back to ship hoping ERT spam works.

so the gameplay loop hasn’t changed from the TM.

sick

**Comment by Losenis** [comment]
> so if they don't run down the entire hive at shutters open all the overconfident robustos die thinking they can avoid death with their shotgun PAS and the fob falls shortly after with ~20 marines running back to ship hoping ERT spam works.

"First push dies game is over" has been a thing for a while now

**Comment by Wayland-Smithy** [comment]
> "First push dies game is over" has been a thing for a while now

Meanwhile there is a PR up to address the xeno snowball via psy points issue, but this speedy unga one with the opposite effect is getting TM.

**Comment by Pariah919** [comment]
>Seems to make snowballs even worse somehow with light/med armor humans out running most xeno castes now, so if they don't run down the entire hive at shutters open all the overconfident robustos die thinking they can avoid death with their shotgun PAS and the fob falls shortly after with ~20 marines running back to ship hoping ERT spam works.
>
if the marines lost the fob i very much doubt that this pr would've made them lose it any faster or slower

**Comment by ilikesaying** [comment]
memed

**Comment by SplinterGP** [comment]
> 
> 
> I dont know what you are "testing" here considering there are a million other ways you could do this, even without touching code
> it has been tested long enough I dont know what you got out of it or even want with this but its been in long enough

testing balance wise, dindt prove that bad

File: code/__DEFINES/mobs.dm [hunk 3]
```diff
@@ -186,8 +186,8 @@ GLOBAL_LIST_INIT(xenoupgradetiers, list(XENO_UPGRADE_BASETYPE, XENO_UPGRADE_INVA
 #define LIMB_STABILIZED (1<<9) //certain suits will support a broken limb while worn such as the b18
 
 /////////////////MOVE DEFINES//////////////////////
-#define MOVE_INTENT_WALK        0
-#define MOVE_INTENT_RUN         1
+#define MOVE_INTENT_WALK        1
+#define MOVE_INTENT_RUN         0
```
[the remark this answers is not in the document]
Comment (spookydonut) [reply code/__DEFINES/mobs.dm:190]: agreed

File: code/datums/components/stamina_behavior.dm [hunk 4]
```diff
@@ -36,21 +36,21 @@
 	UnregisterSignal(parent, list(COMSIG_MOVABLE_MOVED, COMSIG_LIVING_SET_CANMOVE))
 
 
+/datum/component/stamina_behavior/proc/on_canmove_change(datum/source, canmove)
+	SIGNAL_HANDLER
+	var/mob/living/stamina_holder = parent
+	if(canmove || stamina_holder.m_intent == MOVE_INTENT_WALK)
+		return
+	stamina_holder.toggle_move_intent(MOVE_INTENT_WALK)
```
[the remark this answers is not in the document]
Comment (spookydonut) [reply code/datums/components/stamina_behavior.dm:44]: agreed

**Comment by spookydonut** [comment]
David wants this and pending the changes requested I don't see a reason not to allow it.

File: code/datums/components/stamina_behavior.dm (hunk 4 above)
[the remark this answers is not in the document]
Comment (SplinterGP) [reply code/datums/components/stamina_behavior.dm:44]: above already existed if you look below though, just moved

File: code/__DEFINES/mobs.dm (hunk 3 above)
Comment (SplinterGP) [inline code/__DEFINES/mobs.dm:190]:
```suggestion
#define MOVE_INTENT_WALK        0
#define MOVE_INTENT_RUN         1
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Review by SplinterGP — APPROVED** [review]

**Comment by Dolth** [comment]
Honestly no, this makes everything WAY TOO EASY for marines, easier escape, easier chase, no punish for chasing (stam-neuro), it's just a major buff, really, don't think it's necessary at all...

**Comment by SplinterGP** [comment]
> Honestly no, this makes everything WAY TOO EASY for marines, easier escape, easier chase, no punish for chasing (stam-neuro), it's just a major buff, really, don't think it's necessary at all...

You still stop running if you lose stamina

**Comment by TiviPlus** [comment]
Too good unsuprisingly balance it properly

**Comment by Losenis** [comment]
There's very few ways marines can lose stamina in frequent/noticeable enough quantities compared to running. On top of what was said, it nerfed neurotox massively since now it needs to drain stamina all the way from full, at which time it might have metabolized out in great part.

**Comment by jroinc** [comment]
Fully remove the Stamina buffer. Marines don't need one anymore since running doesn't drain it.

Just leave a base 200 stamina. 1 point of buffer if you need one to allow running.

**Comment by Pariah919** [comment]
>Fully remove the Stamina buffer. Marines don't need one anymore since running doesn't drain it.

>Just leave a base 200 stamina. 1 point of buffer if you need one to allow running.

Marines taking 1 stamina damage and losing all ablilty to run is pretty bad, the original buffer isn't needed though, I'll reduce it and make stamina pool 220, so if you take 30 stam damage you walk but stamcrit is the same.

**Comment by SplinterGP** [comment]
testing with new balance stuff

**Comment by Al-1ce** [comment]
Nerf or not this is still ultimately a terrible addition with zero consideration for how speed absolutely destroys the balance of the game.

**Comment by Losenis** [comment]
Oh right this also tanks defiler's hemodile usefulness since the damage to stamina kinda needs running to have any effect at all. Otherwise you need to kill the guy 5 times before his stamina is actually drained.

**Comment by SplinterGP** [comment]
> 
> 
> Oh right this also tanks defiler's hemodile usefulness since the damage to stamina kinda needs running to have any effect at all. Otherwise you need to kill the guy 5 times before his stamina is actually drained.

what?

**Comment by Losenis** [comment]
>/datum/reagent/toxin/xeno_hemodile //slows by 25% and 50% based on base movement speed of marine with shoes on and deals an additional 20% of damage received as stamina damage

The "and deals an additional 20% of damage received as stamina damage" is rendered redundant since marines won't be spending their stamina by running, which in practice they'd be doing pretty damn hard if they're in a situation where hemodile is used to its strength.

**Comment by Surrealaser** [comment]
This is barely better.

Here's the fundamental problem: you're causing a huge, seismic disruption to the game's balance and doing almost nothing to account and adjust for the fact.

Now a huge chunk of Xenos can be outsped and outDPS, and the only way they stand a fighting chance is by engaging Marines exclusively in uber defensive terrain where they have all the advantages, which is where the xeno META is inevitably going to go; diving and aggroing as a Xeno is nothing short of suicide for most castes now. Effectively marines are made more aggressive at the cost of Xenos being forced to fight extremely defensively.

Further, not only is perma sprint offensively powerful, but it also allows marines to retreat to safely much more easily.

This is a bad change in a vacuum, and basically is either going to result in crushing imbalance or, at best, Xenos which lean into extremely defensive fighting to offset the massive advantages offered by perma sprint.

**Comment by Knuff-les** [comment]
> Effectively marines are made more aggressive at the cost of Xenos being forced to fight extremely defensively.
> 
isn't this the state of the game even without this MR tho? like afaik xenos are, and really should be, on a defensive fight from the jump, unless the copious amount of walls, sticky resin, turrets, traps, etc, were supposed to be for something else? Idk if anyone enjoys making marines stay on FOB without pushing for an hour, and idk if any marines enjoy sitting on FOB for an hour holding left click until getting pushing back up.

**Comment by BraveMole** [comment]
I still like the concept, since stamina is a meh addition.

But this is hell for xeno, and really breaks balance

Maybe a small slowdown on weed?

**Comment by Surrealaser** [comment]
> I still like the concept, since stamina is a meh addition.
> 
> But this is hell for xeno, and really breaks balance
> 
> Maybe a small slowdown on weed?

We already have small slowdown on weed; it's still a disaster.

**Comment by BraveMole** [comment]
Yeah so, another day of that, still a disaster for xeno.

Was worth the try, but this is hell

**Comment by TiviPlus** [comment]
This is still garbage for xeno players

**Comment by DerFlammenwerfer** [comment]
> This is still garbage for xeno players
You could give hunters two abilities that can help with this (a ranged bolas and a stationary snap-trap. Former prevents running and could be cut by a knife or machete almost instantly by activating it in-hand, and latter can be resisted out of in like, 1.5 or 2 seconds but roots the marine on the spot when stepped on and damages the leg by a good amount.), and change neurotoxin to add a minor slowing effect to marines as long as it's in the bloodstream and give neuro spit to both Prae and Spitter as additional options. Xenos now have additional counters to the new mobility, one pertaining to a specific caste, and one that can be performed by _several_ other castes. Problem potentially solved.

**Comment by Losenis** [comment]
You're gonna spend a good amount of time identifying and buffing whatever the infinistam broke over and over rather than simply not having infinistam. Way too much stuff depends on stamina drain by running like speed, chem castes, speed castes, and meme builds like no armor jaeger being made stupidly good. And that's assuming you can identify everything this can end up affecting, because nobody immediately thinks about stuff like defiler hemodile's stamdraining damage chem

**Comment by Lauwma** [comment]
Running is dumb but it indeed prevents marines from just running and outspeeding xenos. In my experience xenos are better in bursts so they need to have a certain control over the pace of a fight in TGMC case, it was the stamina. 
Marines are able to extend the length by a little bit with the help of a sprint but if we were to remove the sprint they would be able to just run at xenos and shoot them infinitely.

Though, running stamina is a dumb mechanic. It doesn't feels good to use, in a fast paced game and it sure needs to change.
If you were to remove the running stamina you would need to compensate Xeno side or to nerf the running speed. 

Maybe disabling sprinting when you are firing a wielded weapon could be a good downside, as I believe that shooting while sprinting isn't really practical in real life and would make the marines chases less op.

**Comment by SplinterGP** [comment]
> 
> 
> You're gonna spend a good amount of time identifying and buffing whatever the infinistam broke over and over rather than simply not having infinistam. Way too much stuff depends on stamina drain by running like speed, chem castes, speed castes, and meme builds like no armor jaeger being made stupidly good. And that's assuming you can identify everything this can end up affecting, because nobody immediately thinks about stuff like defiler hemodile's stamdraining damage chem

it still uses stamina to run though, it just doenst drain it, if youre below the stam needed to run you still cant run, that means those castes are EVEN MORE important

**Comment by SplinterGP** [comment]
doenst seem to be up much for expectation, i guess talk to current PL about this or some other implementation of it.

**Comment by David-Stormwell** [comment]
I'll be worked on later maybe, as I had hoped it brought up some points in gameplay to address.

**Pull request closed without merging**